File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
from collections import deque

def solution(n, m, section):
    init_lst = [1] * n
    for i in section:
        init_lst[i-1] = 0
    
    drawing_queue = deque(init_lst)
    result = 0
    
    while drawing_queue:
        if drawing_queue[0] == 1:
            drawing_queue.popleft()
        else:
            result += 1
            for _ in range(m):
                if drawing_queue: drawing_queue.popleft()
                else: break
            
    return result

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.js
const solution = (n, m, section) => {
    painting_arr = Array(n).fill(1);
    section.forEach((i) => painting_arr[i-1] = 0);
    
    let result = 0;
    let cur_idx = 0;
    
    while (cur_idx < n) {
        if (painting_arr[cur_idx] === 0) {
            result += 1;
            cur_idx += m;
        } else {
            cur_idx += 1;
        }
    }
    return result;
}